src/overlay/OverlayManagerTests.cpp

```diff
@@ -75,29 +77,37 @@ class OverlayManagerTests
     class ApplicationStub : public ApplicationImpl
     {
       public:
-        shared_ptr<OverlayManagerStub> mOverlayManager;
         ApplicationStub(VirtualClock& clock, Config const& cfg)
             : ApplicationImpl(clock, cfg)
         {
-            mOverlayManager = make_shared<OverlayManagerStub>(*this);
-            newDB();
         }
+
         virtual OverlayManagerStub&
         getOverlayManager() override
         {
-            return *mOverlayManager;
+            auto& overlay = ApplicationImpl::getOverlayManager();
+            return dynamic_cast<OverlayManagerStub&>(overlay);
```

Comment:
the use of dynamic_cast makes it look like you don't know the type, I'd prefer to use static_cast instead. On that note I don't even think we need to override that method.